espnet/asr/asr_utils.py

```diff
@@ -284,6 +284,21 @@ def torch_snapshot(trainer):
     return torch_snapshot
 
 
+def torch_snapshot_iter(savefun=torch.save,
+                        filename='snapshot.iter.{.updater.iteration}'):
```

Comment:
I think that it is not necessary to make a new function.
Just reuse `torch_snapshot` is fine.